lib/containers/utils.pm

```diff
@@ -40,27 +40,59 @@ sub test_seccomp {
-        validate_script_output "firewall-cmd --list-interfaces --zone=docker", sub { /docker0/ };
+sub check_containers_firewall {
+    my ($runtime) = @_;
+    record_info "firewall", "Checking that firewall is enabled, properly configured and containers can reach the Internet";
+    if ($runtime eq 'docker') {
+        my $container_name = 'sut_container';
+        my $docker_version = get_docker_version();
+        systemctl('is-active firewalld');
+        my $running = script_output qq(docker ps -q | wc -l);
+        validate_script_output('ip a s docker0', sub { /state DOWN/ }) if $running == 0;
+        # Docker zone is created for docker version >= 20.10 (Tumbleweed), but it
+        # is backported to docker 19 for SLE15-SP3 and for Leap 15.3
+        if (check_runtime_version($docker_version, ">=20.10") || is_sle('>=15-sp3') || is_leap(">=15.3")) {
+            assert_script_run "firewall-cmd --list-all --zone=docker";
+            validate_script_output "firewall-cmd --list-interfaces --zone=docker", sub { /docker0/ };
+        }
+        # Rules applied before DOCKER. Default is to listen to all tcp connections
+        # ex. output: "1           0        0 RETURN     all  --  *      *       0.0.0.0/0            0.0.0.0/0"
+        validate_script_output "iptables -L DOCKER-USER -nvx --line-numbers", sub { /1.+all.+0\.0\.0\.0\/0\s+0\.0\.0\.0\/0/ };
+        assert_script_run "docker run -id --rm --name $container_name -p 1234:1234 " . registry_url('alpine');
+        my $container_ip = container_ip($container_name, 'docker');
+        # Each running container should have added a new entry to the DOCKER zone.
+        # ex. output: "1           0        0 ACCEPT     tcp  --  !docker0 docker0  0.0.0.0/0            172.17.0.2           tcp dpt:1234"
+        validate_script_output "iptables -L DOCKER -nvx --line-numbers", sub { /1.+ACCEPT.+!docker0 docker0.+$container_ip\s+tcp dpt:1234/ };
+        assert_script_run "docker kill $container_name ";
+
+        assert_script_run "sysctl -a | grep '\\.forwarding'";
+        assert_script_run "firewall-cmd --list-all-zones";
+        script_run "docker run --rm " . registry_url('alpine') . " ip a s";
+        script_run "docker run --rm " . registry_url('alpine') . " ip r s";
+
+        # Connectivity check
+        assert_script_run "docker run --rm " . registry_url('alpine') . " wget " . get_var('OPENQA_URL');
+    } elsif ($runtime eq 'podman') {
+        my $container_name = 'sut_container';
+        systemctl('is-active firewalld');
+
+        # cni-podman0 interface is created when running the first container
+        assert_script_run "podman run --rm " . registry_url('alpine');
+        validate_script_output('ip a s cni-podman0', sub { /,UP/ });
+
+        # Cheking rules of specific running container
+        assert_script_run "podman run -id --rm --name $container_name -p 1234:1234 " . registry_url('alpine');
+        assert_script_run "iptables -vnL";
```

Comment:
IIRC `iptables` has been replaced by `nftables` in some products. Idk if we run this module in any of them but keep that in mind.

Reply:
Please see my comment outside of this thread.